### PR TITLE
Disabled time typing requirement on dev

### LINKED ISSUE #3151: Disable 2 hour threshold for leaderboards in dev environments

### PR DESCRIPTION
**Pull request by Miodec**
Closes #3151

### CODE REVIEW

File: backend/src/dal/leaderboards.ts [hunk 1]
```diff
@@ -62,6 +62,14 @@ export async function update(
 }> {
   const str = `lbPersonalBests.${mode}.${mode2}.${language}`;
   const start1 = performance.now();
+
+  const timeTypingMatch = {
+    $gt: 7200,
```
[the remark this answers is not in the document]
Comment (Miodec) [reply backend/src/dal/leaderboards.ts:67]: Probably a better choice yeah. Fixing.

File: backend/src/api/controllers/result.ts [hunk 2]
```diff
@@ -327,7 +327,7 @@ export async function addResult(
     (funbox === "none" || funbox === "plus_one" || funbox === "plus_two") &&
     !bailedOut &&
     !user.banned &&
-    (user.timeTyping ?? 0) > 7200;
+    (process.env.MODE === "dev" ? true : (user.timeTyping ?? 0) > 7200);
```
Comment (Bruception) [inline backend/src/api/controllers/result.ts:330]: `process.env.MODE === "dev" || (user.timeTyping ?? 0) > 7200`

Comment (Miodec) [reply backend/src/api/controllers/result.ts:330]: Done

**Review by Bruception — APPROVED** [review]
lgtm

**Pull request merged**